src/spec-lang/tc/typecheck.ts

```diff
@@ -107,10 +108,38 @@ export type SScope =
     | SForAll
     | VariableDeclarationStatement;
 
-export type STypingCtx = SScope[];
+export interface STypingCtx {
+    type: AnnotationType;
+    target: AnnotationTarget;
+    scopes: SScope[];
+    isOld: boolean;
+}
+
+function addScope(ctx: STypingCtx, scope: SScope): STypingCtx {
+    return { type: ctx.type, target: ctx.target, scopes: [...ctx.scopes, scope], isOld: ctx.isOld };
+}
+
+function oldCtx(ctx: STypingCtx): STypingCtx {
+    return { type: ctx.type, target: ctx.target, scopes: ctx.scopes, isOld: true };
+}
+
+function fromCtx(ctx: STypingCtx, other: Partial<STypingCtx>): STypingCtx {
+    const t: any = {
```

Comment:
Not a blocker, but do we get any issues if use `STypingCtx` type instead of `any`?

Reply:
Yeah - it doesn't type for me. If you can figure out some magic to make it work I'm down?

Reply:
Fixed in [93c526e](https://github.com/ConsenSys/scribble/pull/154/commits/93c526e31480369c29e5ddbdbf8965f5ea7bfd73)